pkgs/by-name/mu/musl/package.nix

```
@@ -0,0 +1,177 @@
+{
+  lib,
+  callPackage,
+  fetchurl,
+  linuxHeaders,
+  stdenv,
+  # Boolean flags
+  hasLinuxHeaders ? true,
+  useBSDCompatHeaders ? true,
+}:
+let
+  _sources = callPackage ./sources.nix { };
+  arch =
+    if stdenv.hostPlatform.isx86_64 then
+      "x86_64"
+    else if stdenv.hostPlatform.isx86_32 then
+      "i386"
+    else null;
+in
+stdenv.mkDerivation (finalAttrs: {
+  inherit (_sources.musl) pname version src;
+
+  env = {
+    NIX_DONT_SET_RPATH = true;
+    CFLAGS = toString ([
+      "-fstack-protector-strong"
+    ] ++ lib.optionals stdenv.hostPlatform.isPower [
+      "-mlong-double-64"
+    ]);
+  };
+
+  configureFlags = [
+    (lib.enableFeature true "shared")
+    (lib.enableFeature true "static")
+    (lib.enableFeature true "debug")
+    (lib.enableFeatureAs true "wrapper" "all")
+    "--syslibdir=${placeholder "out"}/lib"
+  ];
+
+  outputs = [
+    "out"
+    "bin"
+    "dev"
+  ];
+
+  outputLib = "out";
+
+  # Disable auto-adding stack protector flags, so musl can selectively disable
+  # as needed
+  hardeningDisable = [ "stackprotector" ];
+
+  strictDeps = true;
+
+  enableParallelBuilding = true;
+
+  dontDisableStatic = true;
+
+  dontAddStaticConfigureFlags = true;
+
+  separateDebugInfo = true;
+
+  # Let's be friendlier to debuggers/perf tools
+  # Neither force them on, nor force them off
+  postPatch = ''
+    substituteInPlace configure \
+      --replace -fno-unwind-tables "" \
```

```suggestion
      --replace -fno-unwind-tables "" \
```

`--replace` is deprecated